File: an2_sem4/MPP/Homework/sapt14/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/objectprotocol/FilterSpectacoleRequest.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.objectprotocol;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.Constants;

import java.time.LocalDateTime;

public class FilterSpectacoleRequest implements IRequest{
    String day;

    public FilterSpectacoleRequest(LocalDateTime day) {
        this.day = day.format(Constants.DTO_FORMATTER);
    }

    public LocalDateTime getDay() {
        return LocalDateTime.parse(day, Constants.DTO_FORMATTER);
    }
}

File: an2_sem4/MPP/Homework/sapt10/cs/FestivalSellpoint/FestivalSellpoint.Network/ObjectProtocol/FilterSpectacoleRequest.cs
using FestivalSellpoint.Network.Utils;
using System;

namespace FestivalSellpoint.Network.ObjectProtocol
{
    [Serializable]
    public class FilterSpectacoleRequest : IRequest
    {
        private string _Day;
        public FilterSpectacoleRequest(DateTime day) => _Day = day.ToString(Constants.DTODateTimeFormatter);
        public DateTime Day => DateTime.ParseExact(_Day, Constants.DTODateTimeFormatter, null);
    }
}
